### ISSUE TITLE
t2 run doesn't stop with CTRL + C

### ISSUE THREAD

**Issue by Frijol** [tessel/t2-cli#186]
You get this kind of error message

```sh
➜  cat-feeder git:(master) ✗ t2 run test.js --lan
INFO Connecting to Tessel...
INFO Connected over LAN. IP ADDRESS: 192.168.8.116
INFO Writing test.js to RAM on Kelsey (3832.32 kB)...
INFO Deployed.
INFO Running test.js...
^C/Users/frijol/t2-cli/lib/tessel/tessel.js:22
    this.connection.end(function() {
                   ^
TypeError: Cannot read property 'end' of undefined
    at process.endConnection (/Users/frijol/t2-cli/lib/tessel/tessel.js:22:20)
    at process.g (events.js:257:16)
    at emitNone (events.js:72:20)
    at process.emit (events.js:163:7)
    at Signal.wrap.onsignal (node.js:727:46)
```

and your code keeps running until you pull power

**Comment by johnnyman727** [comment]
The error is fixed in #182 but somehow the logic to close the connection was lost on one of these refactors. I'll add it back into #182.

**Comment by johnnyman727** [comment]
Fixed with #182. (But note that this symptom was exposed again by #222)
